---
title: "immune_cell_abundance"
author: "Caroline Wheeler"
date: "9/28/2021"
output: html_document
---

```{r setup, include=FALSE}
library(tidyverse)
library(ggplot2)
library(ggpubr)
library(reshape2)

knitr::opts_chunk$set(echo = TRUE)

setwd(dirname(getwd()))
source("00-paths.R")
```

<!-- read in gene expression data -->
<!-- ```{r} -->
<!-- genes <- read.csv(file.path(paths$box, "tcga_gene-expressions.csv"))  -->
<!-- ``` -->

<!-- create matrix for CIBERSORT -->
<!-- ```{r} -->
<!-- write.table(genes, "/Users/wheeler.1017/Documents/GitHub/exohnsc/exploratory/data/cibersort_exohnsc.txt", sep = "\t", quote = FALSE, row.names = FALSE) -->
<!-- ``` -->

read in CIBERSORT results
```{r}
cells <- read.csv(file.path(paths$github, "CIBERSORTx_Job12_Results.csv")) %>%
  select(-P.value, -Correlation, -RMSE)

clin <- read.csv(file.path(paths$box, "tcga_clinical.csv")) %>%
  select(file_id.expression, file_id.BAM) %>%
  rename(
    'sample' = file_id.BAM,
    'sample1' = file_id.expression,
  )
  
# why is one sample dropped? 
hpv <- read.csv(file.path(paths$github, "HPV-prevalence-per-sample.csv"))
```

fix names
```{r}
cells <- cells %>% 
  mutate(sample1 = substr(Mixture, 2, length(Mixture))) %>%
  mutate(sample1 = gsub(".", "-", sample1, fixed = TRUE))
```

combine + clean
```{r}
cells <- merge(cells, clin)
cells <- merge(cells, hpv) %>%
  select(-sample1, -Mixture)
```

kruskal wallis test results in a table 
```{r}
results <- list()

for(i in colnames(cells[,2:23])){  
  results[[i]] <- kruskal.test(formula(paste(i, "~ HPV.prev")), data = cells)
}

results_table <- do.call(rbind.data.frame, results) %>%
  select(p.value) %>%
  rename("p-value" = p.value)

collabs <- c("Naive B-cells", "Memory B-cells", "Plasma cells", "CD8+ T-cells", "Naive CD4 T-cells", "Memory CD4 T-cells, resting", "Memory CD4 T-cells, activated", "Follicular-helper T-cells", "Tregs", "Gamma delta T-cells", "NK cells, resting", "NK cells, activated", "Monocytes", "M0 Macrophages", "M1 Macrophages", "M2 Macrophages", "Dendritic cells, resting", "Dendritic cells, activated", "Mast cells, resting", "Mast cells, activated", "Eosinophils", "Neutrophils")

rownames(results_table) <- collabs
results
```


subset those with a p-value < 0.05 and CD8+Tcells and Tregs and transform data
```{r}
cells <- cells %>%
  select(HPV.prev, B.cells.naive, T.cells.CD8, T.cells.follicular.helper, T.cells.regulatory..Tregs., Macrophages.M0, )

labs <- c("HPV.prev", "Naive B-cells", "CD8+ T-cells", "Follicular-helper T-cells", "Tregs", "M0 Macrophages")
colnames(cells) <- labs

cells <- melt(cells, id.vars=0:1)

cells <- cells %>%
  mutate(HPV.prev = ifelse(HPV.prev == 0, "HPV negative", "HPV postive"))
```

Boxplot to visualize
```{r}
#hpv.labs <- c("HPV negative", "HPV positive")
#names(hpv.labs) <- c("0", "1")
#bp <- bp + facet_grid(. ~ HPV.prev, labeller = labeller(HPV.prev = hpv.labs)) 

bp <- ggplot(cells, aes(x=variable, y=value)) + geom_boxplot(aes(fill=HPV.prev)) + xlab("") + ylab("Abundance") + labs(fill = "") + theme_bw(base_size = 18) + theme(axis.text.x = element_text(angle = 45, hjust=1))

bp

```
save plot
```{r}
ggsave(filename = "/Users/wheeler.1017/Documents/projects_rando/hnsc_immunecell_boxplot.png", plot = bp, width = 14, height = 10, dpi = 600)
```

