---
title: "Visualize-differential-gene-expression"
author: "Malven"
date: "10/2/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
library(forcats)
library(ggplot2)
library(ggrepel)
library(dplyr)
library("DESeq2") # install DESeq2 from https://bioconductor.org/packages/release/bioc/html/DESeq2.html
```

# Load data
```{r}
data.dir <- "/fs/ess/PAS1695/projects/HNSC/data/drake-output/7-15-2021"

tcga_exp <- read.csv(file.path(data.dir, "7-15-2021_tcga_gene-expressions.csv"), stringsAsFactors = F)
tcga_norm <- read.csv(file.path(data.dir, "7-15-2021_tcga_gene-expressions_norm.csv"), stringsAsFactors = F)
tcga_clin <- read.csv(file.path(data.dir, "7-15-2021_tcga_clinical.csv"), stringsAsFactors = F)
prev <- read.csv("../data/HPV-prevalence-per-sample.csv")

```

# Data preparation 
```{r}
# TCGA clinical table

clin.mod <- tcga_clin %>%
  rename(file_id.BAM = "sample", file_id.expression = "sample.expression") %>%
  left_join(prev) %>%
  select(sample.expression, HPV.prev) %>%
  arrange(sample.expression) %>%
  mutate(HPV_stat = ifelse(HPV.prev == 1, yes = "Pos", no = "Neg")) %>%
  mutate(HPV_stat = fct_relevel(HPV_stat, "Neg")) %>%
  drop_na(HPV_stat)

# TCGA gene expression table

sampleid <-sapply(str_remove_all(colnames(tcga_exp),"X"),"[")
colnames(tcga_exp) <- sampleid

tcga_mat <- tcga_exp %>%
  gather(-Gene.Symbol, key = "sample", value = "counts") %>%
  spread(key = "Gene.Symbol", value = "counts")

tcga_mat$sample <- gsub("\\.", "-", tcga_mat$sample)

tcga_comp <- tcga_mat %>%
  gather(-sample, key = "gene", value = "counts") %>%
  spread(key= "sample", value = "counts") %>%
  select(c("gene", clin.mod$sample.expression)) %>%
  column_to_rownames(var="gene") %>%
  as.matrix()

```

# DESeq Analysis
```{r}

# DDS method

dds <- DESeqDataSetFromMatrix (countData= tcga_comp , colData= clin.mod, design=~HPV_stat)
dds <- DESeq (dds)
resultsNames(dds)

HPV_deseq.genes <- results(dds, name = "HPV_stat_Pos_vs_Neg") %>%
  as.data.frame() %>%
  rownames_to_column(var = "Genes") %>%
  arrange(pvalue)

write.table(HPV_deseq.genes, row.names = F, sep = "\t", quote = F, col.names = T, file = "../data/HPV_deseq.genes.txt")
```

# Volcano Plot & Box Plot
```{r}
# Volcano Plot

hpv.volc.dat <- HPV_deseq.genes %>%
  mutate(diffexp = ifelse(log2FoldChange > 2.5 & pvalue < 0.05, yes = "UP",
                   ifelse(log2FoldChange < -2.5 & pvalue < 0.05, yes = "DOWN", no = NA))) %>%
  mutate(diffexp = ifelse(is.na(diffexp), yes = "NO", no = diffexp))

hpv.volc.dat %>%
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), col = diffexp)) +
  geom_point() +
  geom_text_repel(data=subset(hpv.volc.dat, log2FoldChange < -3.5 | log2FoldChange > 3.5),
                        aes(x = log2FoldChange, y = -log10(pvalue), label = Genes),
                  colour = "black", size = 2.5, fontface= 2, box.padding = 1.0, max.overlaps = Inf) +
  geom_hline(yintercept = 0.05, color = "red", linetype = 2) +
  geom_vline(xintercept = c(-2.5,2.5), color = "red", linetype = 2) +
  scale_color_manual( breaks = c("UP", "DOWN", "NO"),
                      values = c("orange", "purple", "grey"),
                      label  = c("Upstream", "Downstream", "Not Significant"),
                      name   = "Genes")+
  ggtitle("HPV - Positive vs Negative") +
  theme(legend.key.size = unit(0.5, 'cm'),
        legend.title = element_text(size=10),
        legend.text = element_text(size=10)) +
  theme_bw()

ggsave("../figures/volcano_differential-gene-expression.png", width = 7, height = 6, device = "png")


# Here, I wanted to visualize the DGEs in a boxplot instead.
# Boxplot

sig.genes <- HPV_deseq.genes$Genes[1:10] # here, I extracted the top 10 significant genes

# Created a table for boxplot

boxdat <- tcga_norm %>%
  select(sample, all_of(sig.genes)) %>%
  gather(-sample, key = "gene", value = "norm.exp") %>%
  rename(sample = "sample.expression") %>%
  inner_join(clin.mod)

boxdat %>%
  dplyr::rename("HPV_status" = HPV_stat) %>%
  ggplot(aes(x = gene, y = norm.exp)) + 
  geom_boxplot(aes(fill = HPV_status)) +
  theme_bw() +
  theme(axis.text.x= element_text(angle = 45, hjust = 1)) +
  labs(x = "", y = "Normalized Expression")

```


